internal/completions/httpapi/handler.go

```diff
@@ -433,3 +441,93 @@ func newNonStreamingResponseHandler(logger log.Logger, db database.DB, feature t
 		_, _ = w.Write(completionBytes)
 	}
 }
+
+type clientCodyIgnoreCompatibilityError struct {
+	reason     string
+	statusCode int
+}
+
+// clientCodyIgnoreCompatibilityErrorPrefix value is used to identify specific errors in the Cody clients codebases.
+// When changing its value be sure to update the clients code.
+const clientCodyIgnoreCompatibilityErrorPrefix = "ClientCodyIgnoreCompatibilityError"
+
+func (e *clientCodyIgnoreCompatibilityError) Error() string {
+	return fmt.Sprintf("%s: %s", clientCodyIgnoreCompatibilityErrorPrefix, e.reason)
+}
+
+// clientCodyIgnoreVersionConstraint represents client version constraint following the semver spec.
+type clientCodyIgnoreVersionConstraint string
+
+const (
+	vscodeCodyIgnoreVersionConstraint    clientCodyIgnoreVersionConstraint = "> 1.14.0"
+	jetbrainsCodyIgnoreVersionConstraint clientCodyIgnoreVersionConstraint = "> 5.5.5"
+)
+
+func checkClientCodyIgnoreCompatibility(r *http.Request) *clientCodyIgnoreCompatibilityError {
+	// If Cody context filters are not defined on the instance, we do not restrict client version
+	if conf.SiteConfig().CodyContextFilters == nil {
+		return nil
+	}
+
+	clientName := types.CodyClientName(r.URL.Query().Get("client-name"))
+	if clientName == "" {
+		return &clientCodyIgnoreCompatibilityError{
+			reason:     "\"client-name\" query param is required.",
+			statusCode: http.StatusNotAcceptable,
+		}
+	}
+
+	// clientVersionConstraint defines the minimum client version required to support Cody Ignore.
+	type clientVersionConstraint struct {
+		client     types.CodyClientName
+		constraint clientCodyIgnoreVersionConstraint
+	}
+	var cvc clientVersionConstraint
+	switch clientName {
+	case types.CodyClientWeb:
+		return nil
+	case types.CodyClientVscode:
+		cvc = clientVersionConstraint{client: clientName, constraint: vscodeCodyIgnoreVersionConstraint}
+	case types.CodyClientJetbrains:
+		cvc = clientVersionConstraint{client: clientName, constraint: jetbrainsCodyIgnoreVersionConstraint}
+	default:
+		return &clientCodyIgnoreCompatibilityError{
+			reason:     fmt.Sprintf("please use one of the supported clients: %s, %s.", types.CodyClientVscode, types.CodyClientJetbrains),
+			statusCode: http.StatusNotAcceptable,
+		}
+	}
+
+	clientVersion := r.URL.Query().Get("client-version")
+	if clientVersion == "" {
+		return &clientCodyIgnoreCompatibilityError{
+			reason:     "\"client-version\" query param is required.",
+			statusCode: http.StatusNotAcceptable,
+		}
+	}
+
+	c, err := semver.NewConstraint(string(cvc.constraint))
+	if err != nil {
+		return &clientCodyIgnoreCompatibilityError{
+			reason:     fmt.Sprintf("Cody for %s version constraint \"%s\" doesn't follow semver spec.", cvc.client, cvc.constraint),
```

Comment:
FYI `%q` adds quotes for you.

```suggestion
			reason:     fmt.Sprintf("Cody for %s version constraint %q doesn't follow semver spec.", cvc.client, cvc.constraint),
```